tests/share/test_harvest.py

```diff
@@ -401,3 +401,45 @@ def test_harvest_after(self, monkeypatch, now, end_date, harvest_after, should_r
         tasks.harvest()
 
         assert source_config.harvester.get_class()._do_fetch.called == should_run
+
+    def test_latest_date(self):
+        source_config = factories.SourceConfigFactory(
+            full_harvest=True,
+            earliest_date=pendulum.parse('2017-01-01').date()
+        )
+
+        # We have a harvest log with start_date equal to earliest_date
+        # but a different source_config
+        factories.HarvestLogFactory(
+            start_date=pendulum.parse('2017-01-01').date(),
+            end_date=pendulum.parse('2017-01-02').date(),
+        )
+
+        assert len(HarvestScheduler(source_config).all(cutoff=pendulum.parse('2018-01-01').date())) == 365
+
+    def test_caught_up(self):
+        source_config = factories.SourceConfigFactory(
+            full_harvest=True,
+            earliest_date=pendulum.parse('2017-01-01').date()
+        )
+
+        factories.HarvestLogFactory(
+            source_config=source_config,
+            start_date=pendulum.parse('2017-01-01').date(),
+            end_date=pendulum.parse('2017-01-02').date(),
+        )
+
+        factories.HarvestLogFactory(
+            source_config=source_config,
+            start_date=pendulum.parse('2018-01-01').date(),
+            end_date=pendulum.parse('2018-01-02').date(),
+        )
+
+        assert len(HarvestScheduler(source_config).all(cutoff=pendulum.parse('2018-01-01').date())) == 0
```

Comment:
It will fill in gaps, just not if earliest date has a harvestlog. Which is just a really bad way to detect if all the other logs have been made... We should probably just toss a boolean field onto SourceConfig to keep track of that.